### ISSUE TITLE
Long running process and Intialize/Terminate

### ISSUE THREAD

**Issue by byrnedo** [gographics/imagick#123]
I'm using imagick in a long running process that is reading work from a queue.
Right now it does the init/terminate calls in the main function. 
Could this possibly lead to memory leaks?
Should I instead do the init/terminate around each work job?

 I'm seeing quite a lot of leaking (and it could well be down to my code), but I've double checked that I manually destroy everything and that any functions that create new wands don't overwrite existing wand variables.

**Comment by justinfx** [comment]
Have you tried profiling your code to rule out whether the leak is in Go or in C?

The state of the bindings currently don't have any effect to init and terminate ImageMagick more than once. This was done to prevent crashes people were encountering when they were using them around independant workers. Even from the ImageMagick docs one is meant to initialize once for the program and terminate at the end. The terminate does have the effect of cleaning up all that temp garbage that ImageMagick creates. 

Are you able to reduce your issue down to a small working example that shows a memory leak? Also can you include your ImageMagick version?

**Comment by byrnedo** [comment]
Thanks for the quick reply.

OK, Then I know it's not that I need to do the init/terminate differently.

I tried pprof but I didn't have much luck getting heap data from it, I'm going to continue trying to. I also tried running  a benchmark with valgrind and that shows lost mem, but I don't really know if I can trust that.

I'll see if I can come up with an example. There's probably a whole lot more wrong with the code since it's a port from an earlier php version.

From convert, the version is `ImageMagick 6.8.9-9 Q16 x86_64 2017-02-22`.
